tests/instruments/thorlabs/test_mpc320.py

```diff
@@ -0,0 +1,537 @@
+import random
+import unittest
+import unittest.mock
+import qmi
+from qmi.core.exceptions import QMI_InstrumentException
+from qmi.core.transport import QMI_SerialTransport
+from qmi.instruments.thorlabs import Thorlabs_Mpc320
+from qmi.instruments.thorlabs.apt_protocol import (
+    AptChannelJogDirection,
+    AptChannelState,
+)
+
+
+class TestThorlabsMPC320(unittest.TestCase):
+    def setUp(self):
+        # Patch QMI context and make instrument
+        self._ctx_qmi_id = f"test-tasks-{random.randint(0, 100)}"
+        qmi.start(self._ctx_qmi_id)
+        self._transport_mock = unittest.mock.MagicMock(spec=QMI_SerialTransport)
```

Comment:
So, there was this discussion of not using the `qmi.start` in unit-tests anymore. It is indeed unnecessary to open localhost contexts for unit-tests as we actually just need to test the driver. To replace this you can use the `/tests/patcher.py`.

You can either patch the `qmi.start` with
```python
from tests.patcher import PatcherQmiContext
...
        self._ctx_qmi_id = f"test-tasks-{random.randint(0, 100)}"
        self.qmi_patcher = PatcherQmiContext()
        self.qmi_patcher.start(self._ctx_qmi_id)
```
which is patching for this type of tests. For testing the driver class directly you can do
```python
from tests.patcher import PatcherQmiContext as QMI_Context
...
        patcher = patch('qmi.instruments.thorlabs.mpc320.create_transport', spec=QMI_SerialTransport)
        self._transport_mock = patcher.start().return_value
        self.addCleanup(patcher.stop)
        # Make DUT
        self._instr: Thorlabs_Mpc320 = Thorlabs_Mpc320(
            QMI_Context("test_mpc320"), Thorlabs_Mpc320, "serial:transport_str"
        )
        self._instr.open()
```

Reply:
was meaning to change that, thanks for pointing it out. I have still kept the patcher for the transport with the `with` keyword